python_modules/dagster/dagster/core/definitions/run_status_sensor_definition.py

```diff
@@ -546,7 +546,13 @@ def _wrapped_fn(context: SensorEvaluationContext):
                                     update_timestamp=update_timestamp.isoformat(),
                                 ).to_json()
                             )
-                            yield from sensor_return
+
+                            if isinstance(
+                                sensor_return, (RunRequest, SkipReason, PipelineRunReaction)
```

Comment:
can they also return a list of `RunRequest`?

Reply:
i think the `else: yield from sensor_return` should take care of that but i will confirm that constructing a list of run requests and returning the list (rather than yielding multiple run requests) works as expected

Reply:
added a couple more types of run status sensors and confirmed they all work:
* yielding a single run request
* returning a single run request
* yield multiple run requests
* returning a list of run requests